src/libasr/pass/global_stmts.cpp

```diff
@@ -105,57 +105,31 @@ void pass_wrap_global_stmts(Allocator &al,
     }
 
     if (return_var) {
-        // The last item was an expression, create a function returning it
-
         // The last defined `return_var` is the actual return value
         ASR::down_cast2<ASR::Variable_t>(return_var)->m_intent = ASRUtils::intent_return_var;
+    }
 
-
-        ASR::asr_t *fn = ASRUtils::make_Function_t_util(
-            al, loc,
-            /* a_symtab */ fn_scope,
-            /* a_name */ fn_name,
-            nullptr, 0,
-            /* a_args */ nullptr,
-            /* n_args */ 0,
-            /* a_body */ body.p,
-            /* n_body */ body.size(),
-            /* a_return_var */ return_var_ref,
-            ASR::abiType::BindC,
-            ASR::Public, ASR::Implementation,
-            nullptr,
-            false, false, false, false, false,
-            nullptr, 0, nullptr, 0,
-            false, false, false);
-        std::string sym_name = fn_name;
-        if (unit.m_global_scope->get_symbol(sym_name) != nullptr) {
-            throw LCompilersException("Function already defined");
-        }
-        unit.m_global_scope->add_symbol(sym_name, down_cast<ASR::symbol_t>(fn));
-    } else {
-        // The last item was a statement, create a subroutine (returning
-        // nothing)
-        ASR::asr_t *fn = ASRUtils::make_Function_t_util(
-            al, loc,
-            /* a_symtab */ fn_scope,
-            /* a_name */ fn_name,
-            nullptr, 0,
-            /* a_args */ nullptr,
-            /* n_args */ 0,
-            /* a_body */ body.p,
-            /* n_body */ body.size(),
-            nullptr,
-            ASR::abiType::Source,
-            ASR::Public, ASR::Implementation, nullptr,
-            false, false, false, false, false,
-            nullptr, 0, nullptr, 0,
-            false, false, false);
-        std::string sym_name = fn_name;
-        if (unit.m_global_scope->get_symbol(sym_name) != nullptr) {
-            throw LCompilersException("Function already defined");
-        }
-        unit.m_global_scope->add_symbol(sym_name, down_cast<ASR::symbol_t>(fn));
+    ASR::asr_t *fn = ASRUtils::make_Function_t_util(
+        al, loc,
+        /* a_symtab */ fn_scope,
+        /* a_name */ fn_name,
+        nullptr, 0,
+        /* a_args */ nullptr,
+        /* n_args */ 0,
+        /* a_body */ body.p,
+        /* n_body */ body.size(),
+        /* a_return_var */ (return_var ? return_var_ref : nullptr),
+        (return_var ? ASR::abiType::BindC : ASR::abiType::Source),
```

Comment:
This is fine, I think. If a tertiary operator doesn't suit our design, we can use some variable to store the values and pass them here.